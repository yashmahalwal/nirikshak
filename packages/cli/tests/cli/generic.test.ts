import path from "path";
import childProcess from "child_process";
import fs from "fs-extra";
const flags = ["--version", "-v", "--help", "-h"];
const modulePath = path.resolve(__dirname, "../../lib/index.js");
describe("Command output", () => {
    beforeAll(() => {
        if (!fs.existsSync(modulePath))
            childProcess.execSync("cd packages/cli && npm run build:dev");
        process.chdir(__dirname);
    });

    for (const flag of flags)
        test(flag, (done) => {
            expect.hasAssertions();
            const output = childProcess.spawn("node", [modulePath, flag]);

            const chunks: Buffer[] = [];
            output.stdout.on("data", (chunk: Buffer) => {
                chunks.push(chunk);
            });

            output.stdout.on("end", () => {
                expect(Buffer.concat(chunks).toString()).toMatchSnapshot();
                done();
            });
        });
});
