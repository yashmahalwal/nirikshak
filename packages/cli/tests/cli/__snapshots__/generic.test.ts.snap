// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Command output --help 1`] = `
"[97m  _   _   _          _   _            _               _    [39m
[97m | \\\\ | | (_)  _ __  (_) | | __  ___  | |__     __ _  | | __[39m
[97m |  \\\\| | | | | '__| | | | |/ / / __| | '_ \\\\   / _\` | | |/ /[39m
[97m | |\\\\  | | | | |    | | |   <  \\\\__ \\\\ | | | | | (_| | |   < [39m
[97m |_| \\\\_| |_| |_|    |_| |_|\\\\_\\\\ |___/ |_| |_|  \\\\__,_| |_|\\\\_\\\\[39m
[97m                                                           [39m
[4m[1mTest your REST APIs with ease[22m[24m


Usage: index.js <command> [options]

Commands:
  index.js init              Initialise nirikshak in the project
  index.js add <name> [dir]  Add a resource to the project
  index.js remove <name>     Remove a resource from the project
  index.js run [name..]      Run tests for specified resources.

Options:
  --config, -c   path to the configuration file      [default: \\"nirikshak.json\\"]
  --help, -h     Show help                                             [boolean]
  --version, -v  Show version number                                   [boolean]

Examples:
  index.js init                 Initialises the project using configuration in
                                nirikshak.json
  index.js init -c config.json  Initialises the project using configuration in
                                config.json
  index.js add student          Adds files for a resource named student in
                                student directory.
  index.js add student myDir    Adds files for a resource named student in myDir
                                directory.
  index.js remove student       Remove the resource named student.
"
`;

exports[`Command output --version 1`] = `
"[97m  _   _   _          _   _            _               _    [39m
[97m | \\\\ | | (_)  _ __  (_) | | __  ___  | |__     __ _  | | __[39m
[97m |  \\\\| | | | | '__| | | | |/ / / __| | '_ \\\\   / _\` | | |/ /[39m
[97m | |\\\\  | | | | |    | | |   <  \\\\__ \\\\ | | | | | (_| | |   < [39m
[97m |_| \\\\_| |_| |_|    |_| |_|\\\\_\\\\ |___/ |_| |_|  \\\\__,_| |_|\\\\_\\\\[39m
[97m                                                           [39m
[4m[1mTest your REST APIs with ease[22m[24m


0.0.0
"
`;

exports[`Command output -h 1`] = `
"[97m  _   _   _          _   _            _               _    [39m
[97m | \\\\ | | (_)  _ __  (_) | | __  ___  | |__     __ _  | | __[39m
[97m |  \\\\| | | | | '__| | | | |/ / / __| | '_ \\\\   / _\` | | |/ /[39m
[97m | |\\\\  | | | | |    | | |   <  \\\\__ \\\\ | | | | | (_| | |   < [39m
[97m |_| \\\\_| |_| |_|    |_| |_|\\\\_\\\\ |___/ |_| |_|  \\\\__,_| |_|\\\\_\\\\[39m
[97m                                                           [39m
[4m[1mTest your REST APIs with ease[22m[24m


Usage: index.js <command> [options]

Commands:
  index.js init              Initialise nirikshak in the project
  index.js add <name> [dir]  Add a resource to the project
  index.js remove <name>     Remove a resource from the project
  index.js run [name..]      Run tests for specified resources.

Options:
  --config, -c   path to the configuration file      [default: \\"nirikshak.json\\"]
  --help, -h     Show help                                             [boolean]
  --version, -v  Show version number                                   [boolean]

Examples:
  index.js init                 Initialises the project using configuration in
                                nirikshak.json
  index.js init -c config.json  Initialises the project using configuration in
                                config.json
  index.js add student          Adds files for a resource named student in
                                student directory.
  index.js add student myDir    Adds files for a resource named student in myDir
                                directory.
  index.js remove student       Remove the resource named student.
"
`;

exports[`Command output -v 1`] = `
"[97m  _   _   _          _   _            _               _    [39m
[97m | \\\\ | | (_)  _ __  (_) | | __  ___  | |__     __ _  | | __[39m
[97m |  \\\\| | | | | '__| | | | |/ / / __| | '_ \\\\   / _\` | | |/ /[39m
[97m | |\\\\  | | | | |    | | |   <  \\\\__ \\\\ | | | | | (_| | |   < [39m
[97m |_| \\\\_| |_| |_|    |_| |_|\\\\_\\\\ |___/ |_| |_|  \\\\__,_| |_|\\\\_\\\\[39m
[97m                                                           [39m
[4m[1mTest your REST APIs with ease[22m[24m


0.0.0
"
`;
